pycytominer/cyto_utils/cells.py

```diff
@@ -644,12 +645,14 @@ def _compartment_df_generator(
 
     def merge_single_cells(
         self,
-        compute_subsample=False,
-        sc_output_file="none",
-        compression_options=None,
-        float_format=None,
-        single_cell_normalize=False,
-        normalize_args=None,
+        compute_subsample: bool = False,
```

Comment:
We haven't used this function argument declaration syntax in pycytominer, historically. I'd love to move towards it, but I wonder if it's separate pull request to do so. In other words, is there a downside to having mixed function argument declarations?

Reply:
My understanding from the Python docs and in practice is that including type hints is mostly for developer courtesy (increasing code readability/understandability) and potentially linting alerts/blocks (for example, with [Mypy](http://mypy-lang.org/)). See this note from [`typing`](https://docs.python.org/3/library/typing.html#module-typing):
> Note: The Python runtime does not enforce function and variable type annotations.

Adding to these thoughts: there are some references in the wild about "Gradual Typing" which could be adopted here ([example presentation on this topic](https://www.youtube.com/watch?v=zKre4DKAB30&t=1111s&ab_channel=PyCascades)). It may come down to feelings/attitude towards dynamic vs static typing and community developer comfort level.

I made these changes because I was focused on this method already and so as to assist Pycytominer developers in their future work here 🙂 . To your point and if you'd like, we could redact the type hinting changes for this PR and move forward on that work within a dedicated issue (maybe this should happen either way?).

Reply:
Thanks for linking the video! Yes, let's embrace gradual typing. I will keep this in mind in all future pull request reviews.

Reply:
Sounds great! Thank you.